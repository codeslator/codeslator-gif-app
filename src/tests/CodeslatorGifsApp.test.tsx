import { shallow } from 'enzyme';
import { CodeslatorGifApp } from '../CodeslatorGifsApp';

describe('CodesGifApp Component', () => {


  test('Should to show the compoenent correctly', () => {
    const wrapper = shallow(<CodeslatorGifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should to show a list of categories', () => {
    const categories = ['Pokemon', 'Naruto'];
    const wrapper = shallow(<CodeslatorGifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
})