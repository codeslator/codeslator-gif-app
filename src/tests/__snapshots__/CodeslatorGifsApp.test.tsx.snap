// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodesGifApp Component Should to show a list of categories 1`] = `
<Fragment>
  <div
    className="container mt-3"
  >
    <h1>
      CodesGifs App
    </h1>
    <AddCategory
      categories={
        Array [
          "Pokemon",
          "Naruto",
        ]
      }
      setCategories={[Function]}
    />
    <div
      className="mt-3"
    >
      <GifGrid
        category="Pokemon"
        key="0"
      />
      <hr />
      <GifGrid
        category="Naruto"
        key="1"
      />
      <hr />
    </div>
  </div>
</Fragment>
`;

exports[`CodesGifApp Component Should to show the compoenent correctly 1`] = `
<Fragment>
  <div
    className="container mt-3"
  >
    <h1>
      CodesGifs App
    </h1>
    <AddCategory
      categories={Array []}
      setCategories={[Function]}
    />
    <div
      className="mt-3"
    />
  </div>
</Fragment>
`;
